src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,565 @@
+
+            # Change into so directory to ensure threshold files can be found
+            rtlsim_so = self.get_nodeattr("rtlsim_so")
+            so_dir = os.path.dirname(os.path.realpath(rtlsim_so))
+            olcwd = os.getcwd()
+            os.chdir(so_dir)
+            num_out_values = self.get_number_output_values()
+            reset_rtlsim(sim)
+            total_cycle_count = rtlsim_multi_io(
+                sim,
+                io_dict,
+                num_out_values,
+                trace_file=trace_file,
+                sname=sname,
+                liveness_threshold=pyverilate_get_liveness_threshold_cycles(),
+            )
+            self.set_nodeattr("cycles_rtlsim", total_cycle_count)
+            os.chdir(olcwd)
+            output = io_dict["outputs"][ostream_name]
+
+            # Manage output data
+            odt = self.get_output_datatype()
+            target_bits = odt.bitwidth()
+            packed_bits = self.get_outstream_width()
+            out_npy_path = "{}/output.npy".format(code_gen_dir)
+            out_shape = self.get_folded_output_shape()
+
+            rtlsim_output_to_npy(output, out_npy_path, odt, out_shape, packed_bits, target_bits)
+
+            # load and reshape output
+            output = np.load(out_npy_path)
+            oshape = self.get_normal_output_shape()
+            output = np.asarray([output], dtype=np.float32).reshape(*oshape)
+            context[node.output[0]] = output
+        else:
+            raise Exception(
+                """Invalid value for attribute exec_mode! Is currently set to: {}
+            has to be set to one of the following value ("cppsim", "rtlsim")""".format(
+                    mode
+                )
+            )
+
+    def code_generation_ipi(self):
+        """Constructs and returns the TCL commands for node instantiation as an RTL
+        block."""
+        rtl_file_list = [
+            x.replace("template", self.onnx_node.name) for x in self.get_rtl_file_list()
+        ]
+        code_gen_dir = self.get_nodeattr("code_gen_dir_ipgen")
+        source_target = "./ip/verilog/rtl_ops/%s" % self.onnx_node.name
+        cmd = ["file mkdir %s" % source_target]
+
+        for rtl_file in rtl_file_list:
+            cmd.append(
+                "add_files -copy_to %s -norecurse %s"
+                % (source_target, os.path.join(code_gen_dir, rtl_file))
+            )
+
+        # Create an RTL block, not an IP core (-type ip)
+        cmd.append(
+            "create_bd_cell -type module -reference %s %s"
+            % (self.get_nodeattr("gen_top_module"), self.onnx_node.name)
+        )
+
+        return cmd
+
+    def get_verilog_top_module_intf_names(self):
+        intf_names = super().get_verilog_top_module_intf_names()
+        if self.get_nodeattr("runtime_writeable_weights") == 1:
+            intf_names["axilite"] = ["s_axilite"]
+
+        return intf_names
+
+    def get_dynamic_config(self, weights, address_stride=1):
```

Comment:
I do not see this function to be used, could that be used by one of the test cases or is that obsolete?

Reply:
deleted

Reply:
I see that you have deleted this function now, it was the only function that used `find_next_power_of_2(n)` from `finn.util.basic`, can that function with the corresponding test also be deleted then?

Reply:
yup. `get_dynamic_config` was used in pytest to write the threshold weights to the correct addresses. 
this is not necessary anymore, since `make_weight_files` will generate correct pattern with paddings for unused addresses.